Include and expose TIFFSNPrintDirectory() from LibTIFF's tif_sprint.c.
Fix "strip chopping" configuration option for CMake; fixed in LibTIFF
4.4.0
[https://gitlab.com/libtiff/libtiff/-/commit/10921ed8bdfa9346cc4d60e2c2fae159dcbcd19b].
--- a/libtiff/CMakeLists.txt
+++ b/libtiff/CMakeLists.txt
@@ -87,6 +87,7 @@
   tif_predict.c
   tif_print.c
   tif_read.c
+  tif_sprint.c
   tif_strip.c
   tif_swab.c
   tif_thunder.c
--- a/libtiff/libtiff.def
+++ b/libtiff/libtiff.def
@@ -93,6 +93,7 @@
 	TIFFRegisterCODEC
 	TIFFReverseBits
 	TIFFRewriteDirectory
+	TIFFSNPrintDirectory
 	TIFFScanlineSize
 	TIFFScanlineSize64
 	TIFFSetClientInfo
--- a/libtiff/tiffconf.h.cmake.in
+++ b/libtiff/tiffconf.h.cmake.in
@@ -98,7 +98,7 @@
 
 /* Support strip chopping (whether or not to convert single-strip uncompressed
    images to mutiple strips of ~8Kb to reduce memory usage) */
-#cmakedefine STRIPCHOP_DEFAULT 1
+#cmakedefine STRIPCHOP_DEFAULT TIFF_STRIPCHOP
 
 /* Enable SubIFD tag (330) support */
 #cmakedefine SUBIFD_SUPPORT 1
